admin/services/check-form.service.js

```diff
@@ -126,6 +126,24 @@ const checkFormService = {
       const promises = modifiedCheckWindows.map(cw => { cw.save() })
       return Promise.all(promises)
     }
+  },
+  /**
+   * Append check windows name(s) and canDelete to check form object.
+   * @param checkWindows
+   * @returns {Promise.<void>}
+   */
+  objectFormattingForSinglePage: async (checkWindows) => {
```

Comment:
The name of this method still leaves me wondering what it does.

Reply:
@jon-shipley It deals with the logic to produce two items of this object, I don't know how to call it! I'll think of something better

Reply:
@pris54 - there is a code smell here.  You have 'and' in the jsdoc description...

I feel that there should be a canDelete() method that only determines whether the checkForm can be deleted.  If you refactored this maybe the remainder could be moved elsewhere.

Reply:
Also - this method is marked as `async` but I think it is synchronous isn't it?

Reply:
@jon-shipley I'm splitting the method into two for clarity, one will produce the window name(s), the other one outputs a true or false for 'canDelete'